src/k8s/vmBuilder.js

```diff
@@ -18,9 +18,9 @@ export const addDisk = (vm, defaultDisk, storage, getSetting) => {
   };
   if (storage.isBootable) {
     const imageSource = getSetting(PROVISION_SOURCE_TYPE_KEY);
-    diskSpec.bootOrder = imageSource === PROVISION_SOURCE_PXE ? BOOT_ORDER_SECOND : BOOT_ORDER_FIRST;
+    diskSpec.bootOrder = imageSource === PROVISION_SOURCE_PXE ? assignBootOrderIndex(vm) : BOOT_ORDER_FIRST;
   } else {
-    delete diskSpec.bootOrder;
```

Comment:
I think this should not include cloud init disks.

Reply:
Done!